plugin/storage/badger/options.go

```diff
@@ -0,0 +1,124 @@
+	"flag"
+	"time"
+
+	"github.com/spf13/viper"
+)
+
+// Options store storage plugin related configs
+type Options struct {
+	primary *NamespaceConfig
+	others  map[string]*NamespaceConfig
+}
+
+// NamespaceConfig is badger's internal configuration data
+type NamespaceConfig struct {
+	namespace      string
+	SpanStoreTTL   time.Duration
+	ValueDirectory string
+	KeyDirectory   string
+	Ephemeral      bool // Setting this to true will ignore ValueDirectory and KeyDirectory
+	SyncWrites     bool
+}
+
+const (
+	suffixKeyDirectory   = ".directory-key"
+	suffixValueDirectory = ".directory-value"
+	suffixEphemeral      = ".ephemeral"
+	suffixSpanstoreTTL   = ".span-store-ttl"
+	suffixSyncWrite      = ".consistency"
+)
+
+// NewOptions creates a new Options struct.
+func NewOptions(primaryNamespace string, otherNamespaces ...string) *Options {
+	options := &Options{
+		primary: &NamespaceConfig{
+			namespace:      primaryNamespace,
+			SpanStoreTTL:   time.Hour * 72, // Default is 3 days
+			SyncWrites:     false,          // Performance over consistency
+			Ephemeral:      true,           // Default is ephemeral storage
+			ValueDirectory: "",
+			KeyDirectory:   "",
+		},
+		others: make(map[string]*NamespaceConfig, len(otherNamespaces)),
+	}
+
+	for _, namespace := range otherNamespaces {
+		options.others[namespace] = &NamespaceConfig{namespace: namespace}
+	}
+
+	return options
+}
+
+// AddFlags adds flags for Options
+func (opt *Options) AddFlags(flagSet *flag.FlagSet) {
+	addFlags(flagSet, opt.primary)
+	for _, cfg := range opt.others {
+		addFlags(flagSet, cfg)
+	}
+}
+
+func addFlags(flagSet *flag.FlagSet, nsConfig *NamespaceConfig) {
+	flagSet.Bool(
+		nsConfig.namespace+suffixEphemeral,
+		nsConfig.Ephemeral,
+		"Mark this storage ephemeral, data is stored in tmpfs (in-memory). Default is true.",
+	)
+	flagSet.Duration(
+		nsConfig.namespace+suffixSpanstoreTTL,
+		nsConfig.SpanStoreTTL,
+		"time.Duration to store the data. Default is 72 hours (3 days).",
+	)
+	flagSet.String(
+		nsConfig.namespace+suffixKeyDirectory,
+		nsConfig.KeyDirectory,
+		"Path to store the keys (indexes), this directory should reside in SSD disk. Set ephmeral to false if you want to define this setting.",
+	)
+	flagSet.String(
+		nsConfig.namespace+suffixValueDirectory,
+		nsConfig.ValueDirectory,
+		"Path to store the values (spans). Set ephmeral to false if you want to define this setting.",
+	)
+	flagSet.Bool(
+		nsConfig.namespace+suffixSyncWrite,
+		nsConfig.SyncWrites,
+		"If all writes should be synced immediately. This will greatly reduce write performance and will require fast SSD drives. Default is false.",
+	)
+}
+
+// InitFromViper initializes Options with properties from viper
+func (opt *Options) InitFromViper(v *viper.Viper) {
+	initFromViper(opt.primary, v)
+	for _, cfg := range opt.others {
+		initFromViper(cfg, v)
+	}
+}
+
+func initFromViper(cfg *NamespaceConfig, v *viper.Viper) {
+	cfg.Ephemeral = v.GetBool(cfg.namespace + suffixEphemeral)
```

Comment:
If this is also intended to replace the current in-memory storage, we need a `max-traces` option as well. See #845/#842

Reply:
Do we need a `max-traces` option? or just a strategy for dealing with a full disk?